aircan/dependencies/cloud/warehouse.py

```
@@ -667,6 +667,62 @@ def append_or_replace_flow(
+
+    ensure_target_exists_from_stage(client, target_fqn, stage_fqn)
+    stage_table = client.get_table(stage_fqn)
+    # Add any columns the staged file introduced (new columns can appear anywhere
+    # in the file — they are matched by name, not position).
+    ensure_table_has_fields(client, target_fqn, stage_table.schema)
+
+    col_list = ", ".join(f"`{field.name}`" for field in stage_table.schema)
+    logger.info("Appending into %s (columns: %s)", target_fqn, col_list)
+    client.query(
+        f"INSERT INTO `{target_fqn}` ({col_list}) "
+        f"SELECT {col_list} FROM `{stage_fqn}`"
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)